Add link to ethash-cl where required for windows

LefterisJP opened the pull request:
In a Windows split repos build we are getting unresolved external symbols
for libethash-cl related code and hence the build fails.

With this PR I am adding the ethash-cl dependency where required.

LefterisJP committed:
Add link to ethash-cl where required for windows

In Windows split repos build we are getting unresolved external symbols
for libethash-cl related code and hence the build fails.

With this PR I am adding the ethash-cl dependency where required.

chriseth commented:
Does hat force `ETHASHCL` on for all platforms?

LefterisJP commented:
Hm ... I thought we always build with ethash-cl now anyway? If not then those 3 subprojects of webthree are linking to ethash-cl in all platforms and for some reason only in windows cmake does not properly link to them?

@debris can you shed some light onto this?

chriseth commented:
I am still successfully building without ethash-cl because I do not have the opencl library installed.

LefterisJP commented:
This,  just like the PR I have opened to [libethereum](https://github.com/ethereum/libethereum/pull/6), seems to trigger a build failure only in Windows.

It also builds fine for me in Linux. But when examining the cmake file and the created MSVC studio project I do indeed see that the libraries in both cases are not linked correctly.

I am not sure yet why it works in Linux ...

debris commented:
We shouldn't merge that. @LefterisJP please look at [these lines](https://github.com/ethereum/webthree-helpers/blob/develop/cmake/UseEth.cmake#L19-L39) in UseEth.cmake. If `ETHASHCL` is set to 1, and `OpenCL` is found, `ethash-cl` is linked. If you want to change this behaviour, please change it there, but do not force people to use `OpenCl` :)

LefterisJP commented:
This PR needs merging in some form or another. Changes to this repo's cmake file will be required if this is going to compile in Windows as split repositories, so please add the "got issues" label and not the invalid.

@debris I have seen these lines. This is the reason why I added the requirement and the set(ETHASHCL 1). For some reason in Windows those 3 sub-projects try to link to ethash-cl code and fail. So I added this to make it work.

Any idea why this would be happening?

LefterisJP commented:
I believe I know what happens. For an explanation look at [this](https://github.com/ethereum/webthree-helpers/pull/11) PR.

I think this is a better solution. @debris. If we merge the above PR I linked the problem is solved for when libethereum has been compiled with libethash-cl. If it has not been compiled with `ETHASHCL`  as 1 which is the default then I guess webthree would not even produce these unresolved linker symbols.

LefterisJP commented:
Okay so after a talk with @debris closing this PR since it seems he had missed something which he adds here: https://github.com/ethereum/webthree/pull/34
